src/os/bluestore/BlueStore.cc

```diff
@@ -5526,50 +5526,26 @@ int BlueStore::_open_db(bool create, bool to_repair_db, bool read_only)
 
   if (kv_backend == "rocksdb") {
     options = cct->_conf->bluestore_rocksdb_options;
-
-    if (cct->_conf.get_val<bool>("bluestore_rocksdb_cf")) {
-      /* enable sharding of rocksdb */
-      map<string,string> cf_map;
-      cct->_conf.with_val<string>("bluestore_rocksdb_cfs",
-                                  get_str_map,
-                                  &cf_map,
-                                  " \t");
-      std::map<std::string, size_t> shards {
-        {PREFIX_SUPER, 0},
-        {PREFIX_STAT, 0},
-        {PREFIX_COLL, 0},
-        {PREFIX_OBJ, 0},
-        {PREFIX_OMAP, 0},
-        {PREFIX_PGMETA_OMAP, 0},
-        {PREFIX_DEFERRED, 0},
-        {PREFIX_ALLOC, 0},
-        {PREFIX_ALLOC_BITMAP, 0},
-        {PREFIX_SHARED_BLOB, 0}
-      };
-
-      for (auto& i : cf_map) {
-        std::string prefix = i.first.substr(0, 1);
-        std::string prefix_count_str = i.first.substr(1);
-        int prefix_count = 1;
-        if (prefix_count_str.size() > 0) {
-          prefix_count = std::stoi(prefix_count_str);
-        }
-        auto sh = shards.find(prefix);
-        if (sh == shards.end()) {
-          derr << __func__ << " db prefix '" << prefix << "' illegal " << dendl;
-          return -EINVAL;
-        }
-        sh->second = prefix_count;
-        dout(10) << "prefix '" << prefix << "' split into " << prefix_count << "column families" << dendl;
-
-        for(int k = 0; k < prefix_count; k++) {
-          std::string cf_name = prefix + "-" + to_string(k);
-          dout(10) << "column family name=" << cf_name << ", options=" << i.second << dendl;
-          cfs.push_back(KeyValueDB::ColumnFamily(cf_name, i.second));
-        }
+    if (create) {
+      if (cct->_conf.get_val<bool>("bluestore_rocksdb_cf")) {
+        std::string sharding_schema = cct->_conf.get_val<string>("bluestore_rocksdb_cfs");
+        r = write_meta("sharding", sharding_schema);
+        if (r < 0)
+          return r;
       }
+    }
+    std::string sharding_schema;
+    r = read_meta("sharding", &sharding_schema);
```

Comment:
is it possible to alter per-cf rocksdb options (other than number of shards) without resharding? e.g. if we wanted to tune a given cf for better performance